### PR TITLE
[Validator] Validate that apps can import the Pod

### PR DESCRIPTION
**Pull request by segiddins**
Closes https://github.com/CocoaPods/CocoaPods/issues/2095 in a circuitous manner.
Closes https://github.com/CocoaPods/CocoaPods/issues/2134.

- [x] CHANGELOG
- [x] Tests
  - [x] Project generation
  - [x] Updating existing specs
  - [x] Update integration spec
- [x] Consider importing headers when not validating as a framework?

### CODE REVIEW

**Comment by orta** [comment]
clever

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -361,13 +368,53 @@ def download_pod
...
+        import_statement = use_frameworks ? "import #{pod_target.product_module_name}\n" : ''
+        source_file.open('w') { |f| f << import_statement }
+      else
+        source_file = validation_dir.+('App/main.m')
+        source_file.parent.mkpath
+        import_statement = if use_frameworks
+          "@import #{pod_target.product_module_name};\n"
+        else
+          header_name = "#{pod_target.product_module_name}/#{pod_target.product_module_name}.h"
+          if pod_target.sandbox.public_headers.root.+(header_name).file?
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: this logic needs to be extracted somewhere, probably into `PodTarget` so we can also skip creating an umbrella header if it exists?

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -610,18 +657,17 @@ def parse_xcodebuild_output(output)
...
       when :watchos
-        command << ' CODE_SIGN_IDENTITY=- -sdk watchsimulator'
+        command += %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
       when :tvos
-        command << ' CODE_SIGN_IDENTITY=- -sdk appletvsimulator'
+        command += %w(CODE_SIGN_IDENTITY=- -sdk appletvsimulator)
       end
 
-      output, status = _xcodebuild "#{command} 2>&1"
+      output, status = Dir.chdir(validation_dir) { _xcodebuild(command) }
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: we can probably get rid of the `chdir` by passing an absolute path to the workspace?

**Comment by segiddins** [comment]
@CocoaPods/Core please review, as this is a big change

**Comment by segiddins** [comment]
:shipit:

**Comment by neonichu** [comment]
✨ 💖 ✨

**Comment by kylef** [comment]
Looks good to me :+1:

**Pull request merged**